File: Project3_Python/Animals/Wolf.py
from Animal import Animal

from Organism import OrganismsEnum


class Wolf(Animal):
    __DEFAULT_INITIATIVE = 5
    __DEFAULT_STRENGTH = 9

    def __init__(self, world, position, strength=__DEFAULT_STRENGTH):
        super().__init__(world, position, strength)
        self.type_of_organism = OrganismsEnum.Wolf
        self._initiative = self.__DEFAULT_INITIATIVE

    def __str__(self):
        return "Wolf"

File: Project2_Java/src/pl/pg/oop/project2/organisms/animals/Wolf.java
package pl.pg.oop.project2.organisms.animals;

import pl.pg.oop.project2.organisms.Animal;
import pl.pg.oop.project2.organisms.OrganismsEnum;
import pl.pg.oop.project2.organisms.World;

import java.awt.*;

public class Wolf extends Animal {
    private final static int DEFAULT_INITIATIVE = 5;
    private final static int DEFAULT_STRENGTH = 9;
    public Wolf(World world, Point position) {
        super(world, position);
        this.TypeOfOrganism = OrganismsEnum.WOLF;
        this.initiative = DEFAULT_INITIATIVE;
        this.strength = DEFAULT_STRENGTH;
    }

    public Wolf(World world, Point position, int strength) {
        super(world, position, strength);
        this.TypeOfOrganism = OrganismsEnum.WOLF;
        this.initiative = DEFAULT_INITIATIVE;
    }

    @Override
    public String toString() {
        return "Wolf";
    }
}
